api/server/sdk/volume_migrate.go

```diff
@@ -229,18 +230,37 @@ func (s *VolumeServer) filterStatusResponseForPermissions(
 		return resp, nil
 	}
 
-	// get all volumes from single inspect
-	allVols, err := s.driver(ctx).Inspect(allVolIds)
+	// Setup ownership locator
+	userinfo, ok := auth.NewUserInfoFromContext(ctx)
+	if !ok {
+		return resp, nil
+	}
+	groupAcls := make(map[string]api.Ownership_AccessType)
+	for _, g := range userinfo.Claims.Groups {
+		groupAcls[g] = api.Ownership_Read
+	}
+	locator := api.VolumeLocator{
+		VolumeIds: allVolIds,
+		Ownership: &api.Ownership{
+			Owner: userinfo.Username,
+			Acls: &api.Ownership_AccessControl{
+				Groups: groupAcls,
+				Collaborators: map[string]api.Ownership_AccessType{
+					userinfo.Username: api.Ownership_Read,
+				},
+			},
+		},
+	}
+
+	allVols, err := s.driver(ctx).Enumerate(&locator, nil)
```

Comment:
The code should use the sdk Inspect call and let it decide what to do

Reply:
Spoke with Luis and we agreed switching to driver.Enumerate() will have better performance as it will not perform a deep inspect.